### PR TITLE
V6

### PR DESCRIPTION
**Pull request by BuddahLD**

### CODE REVIEW

File: crypto/src/main/java/com/virgilsecurity/sdk/crypto/KeyPairType.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+        return RSA_4096_LENGTH;
+      case RSA_8192:
+        return RSA_8192_LENGTH;
+      default:
+        return -1;
+    }
+  }
+
+  /**
+   *  // TODO add a doc
```
Comment (andrii-iakovenko) [inline crypto/src/main/java/com/virgilsecurity/sdk/crypto/KeyPairType.java]: Add javadoc

File: crypto/src/main/java/com/virgilsecurity/sdk/crypto/KeyPairType.java [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+      case RSA_8192:
+      case SECP256R1:
+        return false;
+      default:
+        throw new IllegalStateException("Impossible to get here. Just removing compiler error.");
+    }
+  }
+
+  /**
+   * // TODO add a doc
```
Comment (andrii-iakovenko) [inline crypto/src/main/java/com/virgilsecurity/sdk/crypto/KeyPairType.java]: Add javadoc

File: crypto/src/main/java/com/virgilsecurity/sdk/crypto/KeyPairType.java [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+      case RSA_8192:
+      case SECP256R1:
+        throw new CryptoException("Key is not compound.");
+      default:
+        throw new IllegalStateException("Impossible to get here. Just removing compiler error.");
+    }
+  }
+
+  /**
+   * // TODO add a doc
```
Comment (andrii-iakovenko) [inline crypto/src/main/java/com/virgilsecurity/sdk/crypto/KeyPairType.java]: Add javadoc

File: crypto/src/test/java/com/virgilsecurity/sdk/crypto/VirgilCryptoTest.java [hunk 4]
```diff
@@ -84,21 +88,32 @@
...
+
+    return values.stream().map(key -> Arguments.of(new VirgilCrypto(key)));
+  }
+
+  private static Stream<Arguments> comparableKeyTypesCryptos() {
+    Set<KeyPairType> values = new HashSet<>(Arrays.asList(KeyPairType.values()));
+    values.remove(KeyPairType.RSA_2048);
+    values.remove(KeyPairType.RSA_4096);
+    values.remove(KeyPairType.RSA_8192);
+    values.remove(KeyPairType.SECP256R1); // FIXME check whether this key type has random signatures with Sergey
```
Comment (andrii-iakovenko) [inline crypto/src/test/java/com/virgilsecurity/sdk/crypto/VirgilCryptoTest.java]: Does it verified?

File: crypto/src/test/java/com/virgilsecurity/sdk/crypto/VirgilCryptoTest.java [hunk 5]
```diff
@@ -157,10 +172,9 @@ public void decrypt_verbose_error(VirgilCrypto crypto) throws VirgilException {
 
     VirgilKeyPair keyPairWrong = crypto.generateKeyPair();
 
-    assertEquals(assertThrows(DecryptionException.class, () -> {
+    String exceptionMessage = assertThrows(DecryptionException.class, () -> {
```
Comment (andrii-iakovenko) [inline crypto/src/test/java/com/virgilsecurity/sdk/crypto/VirgilCryptoTest.java]: exceptionMessage is not verified

File: crypto/src/test/java/com/virgilsecurity/sdk/crypto/VirgilCryptoTest.java [hunk 6]
```diff
@@ -372,16 +386,19 @@ public void importPublicKey(VirgilCrypto crypto) throws CryptoException {
     }
   }
 
-  @SignCryptoTest
+  @ParameterizedTest
+  @MethodSource("comparableKeyTypesCryptos")
   public void sign_stream_compareToByteArraySign(VirgilCrypto crypto) throws CryptoException {
     VirgilKeyPair keyPair = crypto.generateKeyPair();
     byte[] signature = crypto.generateSignature(TEXT.getBytes(), keyPair.getPrivateKey());
+//    byte[] signature2 = crypto.generateSignature(TEXT.getBytes(), keyPair.getPrivateKey());
```
Comment (andrii-iakovenko) [inline crypto/src/test/java/com/virgilsecurity/sdk/crypto/VirgilCryptoTest.java]: Why do we need this check?

File: crypto/src/test/java/com/virgilsecurity/sdk/crypto/VirgilCryptoTest.java (hunk 4 above)
Comment (BavovnaIsTaken) [reply crypto/src/test/java/com/virgilsecurity/sdk/crypto/VirgilCryptoTest.java]: yep. forgot to remove fixme

File: crypto/src/test/java/com/virgilsecurity/sdk/crypto/VirgilCryptoTest.java (hunk 6 above)
Comment (BavovnaIsTaken) [reply crypto/src/test/java/com/virgilsecurity/sdk/crypto/VirgilCryptoTest.java]: We had a bug when signature of stream and signature of byte[] were different

File: crypto/src/test/java/com/virgilsecurity/sdk/crypto/VirgilCryptoTest.java (hunk 5 above)
Comment (BavovnaIsTaken) [reply crypto/src/test/java/com/virgilsecurity/sdk/crypto/VirgilCryptoTest.java]: we don't need this test any more at all. We don't provide custom message anymore, we're just using the one that comes from the low-level crypto

**Review by andrii-iakovenko — APPROVED** [review]

**Pull request merged**